test/e2e/lifecycle_test.go

```diff
@@ -0,0 +1,112 @@
+//+build e2e
+
+package e2e
+
+import (
+	"github.com/cardil/operator-e2e-harness/pkg/config"
+	harness "github.com/cardil/operator-e2e-harness/pkg/test"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	eventingv1alpha1 "github.com/openshift-knative/knative-eventing-operator/pkg/apis/eventing/v1alpha1"
+	"github.com/openshift-knative/knative-eventing-operator/pkg/controller/knativeeventing"
+	"github.com/openshift-knative/knative-eventing-operator/test"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+var eventingCR = eventingv1alpha1.SchemeGroupVersion.
+	WithResource("knativeeventings")
+
+func TestBasicLifecycle(t *testing.T) {
+	ctx := test.NewContext(t)
+	defer ctx.Cleanup()
+
+	t.Run("deploy eventing operator", func(t *testing.T) {
+		packageName := "knative-eventing-operator"
+		subscriptionName := "eventing-operator-subscription"
+		_, err := harness.WithOperatorReady(ctx.Base, subscriptionName, packageName)
+		if err != nil {
+			t.Fatal("Failed", err)
+		}
+	})
+
+	t.Run("wait until eventing cr reports ready", func(t *testing.T) {
+		// FIXME: SRVKE-252 Eventing CR should'n be automatically created by operator
+
+		var cr *unstructured.Unstructured
+		var err error
+
+		waitErr := wait.PollImmediate(config.Polling.Interval, config.Polling.Timeout, func() (bool, error) {
+			ns := knativeeventing.Operand
+			cr, err = ctx.Base.Clients.Dynamic.Resource(eventingCR).Namespace(ns).
```

Comment:
I remember discussion where we spoke about generating the client API similar to upstream: https://github.com/knative/eventing-operator/tree/master/pkg/client/clientset/versioned . Then we wouldn't have to use `Dynamic` API.

Reply:
I'm all for client gen especially it is as simple as adding `+client-gen` to files.